Make fails with GCC-15/16

Hi all,

I seem to be having an issue compiling with GCC-15/16. Both give the same issue - I initially tried 16 as that's my system default, but then after seeing a [commit](https://github.com/ps2dev/ps2sdk/pull/767) for gcc-15 support I gave that a shot. Both give the issue:

```
[tom@tom-xps15 ps2sdk]$ make CC=gcc-15

...

src/romimg.c: In function ‘AddFile’:
src/romimg.c:471:27: error: initialization discards ‘const’ qualifier from pointer target type [-Werror=discarded-qualifiers]
  471 |                 char* T = strrchr(fname, '.');
      |                           ^~~~~~~
cc1: all warnings being treated as errors

```

I've managed to get it moving by running `make cc=GCC-11`, but, again, it appears from the above PR that up to at least GCC-15 is supported. I'm very possibly missing something, mea culpa if so, I'm a touch rusty in terms of playing with GCC.

--- a/tools/adpenc/src/adpcm.c
+++ b/tools/adpenc/src/adpcm.c
@@ -204,8 +204,8 @@ static void find_predict( short *samples, double *d_samples, int *predict_nr, in
             if ( fabs( ds ) > max[i] )
                 max[i] = fabs( ds );
 //                printf( "%+5.2f\n", s2 );
-                s_2 = s_1;                                  // new s[t-2]
-                s_1 = s_0;                                  // new s[t-1]
+            s_2 = s_1;                                  // new s[t-2]
+            s_1 = s_0;                                  // new s[t-1]
         }
 
         if ( max[i] < min ) {

--- a/tools/romimg/src/romimg.c
+++ b/tools/romimg/src/romimg.c
@@ -467,9 +467,9 @@ int AddFile(ROMIMG *ROMImg, const char *path, int upperconv)
             tbuf[sizeof(tbuf) - 1] = '\0';
             if (tbuf[0] != '\0') {
                 upperbuff(tbuf);
-                fname = tbuf;
-                char* T = strrchr(fname, '.');
+                char* T = strrchr(tbuf, '.');
                 if (T != NULL) *T = '\0'; //null terminate extension
+                fname = tbuf;
             }
         }
 		int size;

